packages/ui/src/components/Validators.tsx

```diff
@@ -1,61 +1,187 @@
+        {title}
+      </a>
+    </li>
+  );
+};
+
+const navContent = (isActive: boolean, tabName: string, dataTable: any) => {
+  return (
+    <div
+      key={tabName}
+      className={`tab-pane ${isActive ? 'show active' : ''}`}
+      id={tabName}
+      role="tabpanel"
+    >
+      {dataTable}
+    </div>
+  );
+};
 
-  refresh(): void {
+@observer
+export default class Validators extends RefreshableComponent<Props, {}> {
+  async refresh() {
     this.props.validatorsContainer.refresh();
-    this.props.validatorsContainer.toggleableSubscriber.setUpSubscriber();
   }
 
   render() {
+    // If data is not there, display loader.
+    let data = this.props.validatorsContainer.validatorsInfo;
+    if (!data) {
+      return <Loading />;
+    }
+
+    // Contaners for tabs.
+    let navs = [];
+    let tables = [];
+
+    // Prepare bids navigation element.
+    navs.push(
+      navElement(true, 'validators-bids-nav', 'validators-bids-tab', 'Bids')
+    );
+
+    // Prepare bids rows.
+    let rows = Object.keys(data.bids)
+      .map(validatorId => {
+        return {
+          validatorId: validatorId,
+          delegation_rate: data?.bids[validatorId].delegation_rate,
+          stakeStr: data?.bids[validatorId].staked_amount,
+          stakeNum: BigNumber.from(data?.bids[validatorId].staked_amount)
+        };
+      })
+      .sort((a, b) => compareBigNumbers(a.stakeNum, b.stakeNum));
+
+    // Build bids data table.
+    tables.push(
+      navContent(
+        true,
+        'validators-bids-tab',
+        <DataTable
+          title="Bids"
+          headers={['Validator ID', 'Slot', 'Delegation Rate', 'Stake']}
+          rows={rows}
+          renderRow={(bidInfo, index) => {
+            let key = `bids-${bidInfo.validatorId}`;
+            return (
+              <tr key={key}>
+                <td>{bidInfo.validatorId}</td>
+                <td>{index! + 1}</td>
+                <td>{bidInfo.delegation_rate}</td>
+                <td>
+                  <CSPR motes={bidInfo.stakeStr} />
+                </td>
+              </tr>
+            );
+          }}
+          noHeader={true}
+        />
+      )
+    );
+
+    // For each era build navigation element and data table.
+    for (const [index, eraId] of Object.keys(data.era_validators).entries()) {
+      let eraName = `Era ${eraId} ${index == 0 ? '(current)' : ''}`;
+      let tabName = `validator-tab-${eraId}`;
+      let navName = `${tabName}-nav`;
+
+      // Build era navigation element.
+      navs.push(navElement(false, navName, tabName, eraName));
+
+      // Prepare era rows.
+      let rows = Object.keys(data.era_validators[eraId])
+        .map(validatorId => {
+          return {
+            validatorId: validatorId,
+            stakeStr: data?.era_validators[eraId][validatorId],
+            stakeNum: BigNumber.from(data?.era_validators[eraId][validatorId])
+          };
+        })
+        .sort((a, b) => compareBigNumbers(a.stakeNum, b.stakeNum));
```

Comment:
Maybe you could use `.sort((a, b) => a.stakeNum.lt(b.stakeNum))` directly?

Reply:
`.sort` requires -1, 0 or 1 to be returned. `lt` returns boolean.